src/parser/scanner.lex

```diff
@@ -404,7 +404,8 @@ LABEL_FULL_WIDTH            {CN_EN_FULL_WIDTH}{CN_EN_NUM_FULL_WIDTH}*
 
 {DEC}*\.{DEC}+              |
 {DEC}+\.{DEC}*              |
-{DEC}*\.{DEC}*{EXP}         |
+{DEC}*\.{DEC}+{EXP}         |
```

Comment:
I think you could add `{EXP}?` directly.

Reply:
nice!